sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -1031,6 +1028,25 @@ private async Task ReconcileConcurrencyAsync()
                 }
             }
 
+            if (IsSessionProcessor)
+            {
+                int maxAcceptSessions = Math.Min(maxConcurrentCalls, 2 * Environment.ProcessorCount);
```

Comment:
Should we consider calculating `2 * Environment.ProcessorCount` in the constructor and caching?   It won't change and does require an unmanaged call to query _([src](https://source.dot.net/#System.Private.CoreLib/src/System/Environment.CoreCLR.cs,87))_.

That said, this is also not a performance-critical path, so perhaps its not worth the effort.

Reply:
Why do we have the `2 * Environment.ProcessorCount` fallback?

Reply:
This was used in the T1 SDK and we carried it forward.

Reply:
![https://media0.giphy.com/media/xVIkfXYGTJeZKilg3p/200w.gif?cid=82a1493bcqhgv6dig1ke1rbwhdfcphyns0ux00k9uzrqujql&rid=200w.gif&ct=g](https://media0.giphy.com/media/xVIkfXYGTJeZKilg3p/200w.gif?cid=82a1493bcqhgv6dig1ke1rbwhdfcphyns0ux00k9uzrqujql&rid=200w.gif&ct=g)

Reply:
The idea is that you don't spam the service with accept session requests all at once. Imagine someone put 1000 as the number of max concurrent sessions, that means eventually we can have 1000 concurrent sessions being processed, but we don't want to initiate this processing all at the same time. I think the processor count is just a somewhat arbitrary heuristic to achieve the goal of throttling the rate of requests to the service here.